docs/maui/views/popup/popup-result.md

```diff
@@ -0,0 +1,160 @@
+---
+title: Popup<T> - .NET MAUI Community Toolkit
+author: bijington
+description: The PopupService provides a mechanism for displaying Popups within an application using the MVVM pattern.
+ms.date: 05/19/2025
+---
+
+The .NET MAUI Community Toolkit provides the ability to show a [Popup](../Popup.md) to a user. A common more complex scenario is to display a `Popup` and await for a result to be returned when that `Popup` is dismissed. This page covers how the `Popup<T>` class can be used to achieve the desired behavior.
+
+## Building a Popup
+
+A `Popup<T>` can be created in XAML or C# as follows:
+
+### XAML
+
+The following section covers how to create a `Popup<T>` using XAML.
+
+#### Defining your Popup
+
+The easiest way to create a `Popup<T>` is to add a new `.NET MAUI ContentView (XAML)` to your project, this will create 2 files; a _*.xaml_ file and a _*.xaml.cs_ file. The contents of each file can be replaced with the following:
+
+##### .xaml
+
+```xaml
+<toolkit:Popup
+    xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
+    xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
+    xmlns:toolkit="http://schemas.microsoft.com/dotnet/2022/maui/toolkit"
+    x:TypeArguments="system:Boolean"
+    x:Class="MyProject.ReturnResultPopup">
+
+    <VerticalStackLayout>
+        <Label Text="This is a very important message! Do you agree?" />
+
+        <Button Text="Yes" 
+                Clicked="OnYesButtonClicked" />
+
+        <Button Text="No"
+                Clicked="OnNoButtonClicked" />
+    
+    </VerticalStackLayout>
+    
+</toolkit:Popup>
+```
+
+##### .xaml.cs
+
+```csharp
+public partial class ReturnResultPopup : ContentView
```

Comment:
In xaml we use `toolkit:Popup`, should the code-behind follow the same type or omit it?